Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -451,19 +465,28 @@ void CGPathAddRect(CGMutablePathRef path, const CGAffineTransform* transform, CG
     CGPathAddLineToPoint(path, transform, CGRectGetMaxX(rect), CGRectGetMaxY(rect));
     CGPathAddLineToPoint(path, transform, CGRectGetMinX(rect), CGRectGetMaxY(rect));
     CGPathCloseSubpath(path);
+    path->SetLastTransform(transform);
 }
```

Comment:
nit: might want to document:

"We do not set the current point here as with the completion of the rectangle we have not moved."